### DISCUSSION TITLE
Error tcp://192.168.0.200:502: Exception Response(131, 3, IllegalAddress)

### DISCUSSION THREAD

**Discussion by Andrewletsdig18** [nathanmarlor/foxess_modbus#675] [Support] [answered]
Have rs485 conection to ess inverter H1 , using wavershare rs4855 to ethernet converter
introgaton errors after a minuite with
 Error reading registers. Type: RegisterType.HOLDING; start: 36001; count: 3; slave: 247 from tcp://192.168.0.200:502: Exception Response(131, 3, IllegalAddress)

**Comment by FozzieUK** [comment]
I'm assuming the H1 is a G2 ?

The integration is failing to read the firmware versions, which suggests that you may be running a very old version of firmware which doesn't support these modbus registers - can your check and copy below your Master, Manager versions please

**Comment by Andrewletsdig18** [comment]
Dave
Thanks for replying
the inverter is H1-6.0-E.G2
Master is 1.14
Slave is 1.02
Com is 1.17

**Reply by FozzieUK** [reply, marked as answer]
Ok, those are very old versions and there was quite a big change in the modbus registers on more recent versions - latest are Master 1.27 and Manager 1.44

Can you ask your installer for a firmware update, or email fox service with your inverter serial number - they are usually very quick to respond.

Once updated the integration should connect fine 👍

**Comment by Andrewletsdig18** [comment]
OK,  have allready requested the updateThought it might have been causeRegards And  thanksAndrewSent from my Galaxy
-------- Original message --------From: Dave Foster ***@***.***> Date: 12/09/2024  21:00  (GMT+00:00) To: nathanmarlor/foxess_modbus ***@***.***> Cc: Andrewletsdig18 ***@***.***>, Author ***@***.***> Subject: Re: [nathanmarlor/foxess_modbus] Error tcp://192.168.0.200:502: Exception Response(131, 3, IllegalAddress) (Discussion #675) 
Ok, those are very old versions and there was quite a big change in the modbus registers on more recent versions - latest are Master 1.27 and Manager 1.44
Can you ask your installer for a firmware update, or email fox service with your inverter serial number - they are usually very quick to respond.
Once updated the integration should connect fine 👍
